bootstore/src/schemes/v0/peer.rs

```diff
@@ -448,6 +522,108 @@ impl Node {
                     let _ = handle.tx.send(MainToConnMsg::Close).await;
                 }
             }
+            NodeApiRequest::UpdateNetworkConfig { config, responder } => {
+                let current_gen =
+                    self.network_config.as_ref().map_or(0, |c| c.generation);
+                info!(
+                    self.log,
+                    concat!(
+                        "Attempting to update network config with ",
+                        "generation: {}, current_generation: {}"
+                    ),
+                    config.generation,
+                    current_gen,
+                );
+                if current_gen > config.generation {
+                    error!(
+                        self.log,
+                        concat!(
+                            "Attempted network config update with ",
+                            "stale generation: attemped_update_generation: {}, ",
+                            "current_generation: {}"
+                        ),
+                        config.generation,
+                        current_gen,
+                    );
+                    let _ = responder.send(Err(
+                        NodeRequestError::StaleNetworkConfig {
+                            attempted_update_generation: config.generation,
+                            current_generation: current_gen,
+                        },
+                    ));
+                } else if current_gen == config.generation {
+                    warn!(
+                        self.log,
+                        concat!(
+                            "Not updating network config: generation ",
+                            "{} is current"
+                        ),
+                        current_gen
+                    );
+                } else {
+                    self.network_config = Some(config.clone());
+                    NetworkConfig::save(
+                        &self.log,
+                        self.config.network_config_ledger_paths.clone(),
+                        config,
+                    )
+                    .await;
+                    // Broadacst the updated config. We only broadcast
+                    // when we successfully update it so we don't trigger an
+                    // endless broadcast storm.
```

Comment:
I think this will still trigger a small broadcast storm, since each node in the network will receive a new config, update itself, and then call `broadcast_network_config` itself.

Reply:
Yes, that's intentional although I'm not sure it's strictly necessary inside the rack. The goal is to allow all nodes to learn of the update even if there isn't full connectivity due to a network partition. It's cheap though so I kept it for robustness purposes.

Reply:
Agreed on all fronts!